libraries/botbuilder/src/teams/teamsSSOTokenExchangeMiddleware.ts

```diff
@@ -138,16 +140,29 @@ export class TeamsSSOTokenExchangeMiddleware implements Middleware {
         let tokenExchangeResponse: TokenResponse;
         const tokenExchangeRequest: TokenExchangeInvokeRequest = context.activity.value;
 
-        const tokenProvider = ExchangeToken.parse(context.adapter);
+        const tokenProvider = ExchangeToken.safeParse(context.adapter);
 
-        // TODO(jgummersall) convert to new user token client provider when available
         try {
-            tokenExchangeResponse = await tokenProvider.exchangeToken(
-                context,
-                this.oAuthConnectionName,
-                context.activity.channelId,
-                { token: tokenExchangeRequest.token }
+            const userTokenClient = context.turnState.get<UserTokenClient>(
+                (context.adapter as CloudAdapterBase).UserTokenClientKey
             );
+            if (userTokenClient) {
+                tokenExchangeResponse = await userTokenClient.exchangeToken(
+                    context.activity.from.id,
+                    this.oAuthConnectionName,
+                    context.activity.channelId,
+                    { token: tokenExchangeRequest.token }
+                );
+            } else if (tokenProvider.success) {
```

Comment:
Why not use the one from `} else if (ExtendedUserTokenProviderT.check(context.adapter))` (https://github.com/microsoft/botbuilder-js/blob/8ede8316ce82e6a3a1a6cea6fe829f1b2bb16c89/libraries/botbuilder-dialogs/src/prompts/userTokenAccess.ts#L138), instead of using the `safePrse`, it will be less code.

`} else if (ExchangeToken.check(context.adapter)) {`